src/assisted_test_infra/test_infra/controllers/minikube.py

```diff
@@ -0,0 +1,242 @@
+import logging
+import os
+import time
+
+import waiting
+
+
+class MinikubeController:
+    """Manage minikube commands / config changes via k8_client
+    Allow to modify configmap and verify changes
+    Allow to stop/start resources running on minukube plus waiters
+    for actions completed till pods deleted or created.
+    Enable metrics to check memory leaks and cpu usage - TBD
+    In order to access to data resource we:
+    _get_resources_by_kind -> this is the kind of resource
+    _get_obj_resources_by_name -> get the obj resource name
+    """
+
+    CONFIGMAP = "ConfigMap"
+    DEPLOYMENT = "Deployment"
+    REPLICASET = "ReplicaSet"
+    POD = "Pod"
+    SERVICE = "Service"
+
+    def __init__(self, k8_client):
+        """
+        Initialized k8_client accepted from fixture
+        :param k8_client:
+        """
+        self.k8_client = k8_client
+        # mapping for resources kind
+        self.mapping_resources = {
+            self.CONFIGMAP: self._get_resources_by_kind(self.CONFIGMAP, "v1"),
+            self.DEPLOYMENT: self._get_resources_by_kind(self.DEPLOYMENT, "apps/v1"),
+            self.REPLICASET: self._get_resources_by_kind(self.REPLICASET, "apps/v1"),
+            self.POD: self._get_resources_by_kind(self.POD, "v1"),
+            self.SERVICE: self._get_resources_by_kind(self.SERVICE, "v1"),
+        }
+
+    @property
+    def namespace(self):
```

Comment:
Pass those properties to the class constructor, same as for `configmap_name`, `assisted_service_name`  and every env var that is needed within the controller

Reply:
ack , i created properies for each resource and update the mapping , hope that was the meaning.